articles/model-anatomy.adoc

```diff
@@ -0,0 +1,678 @@
+We don't assume any prior knowledge of LionWeb; some experience with general modeling ideas is helpful.
+
+== Introduction
+The fundamental world view in LionWeb is "everything is a model".
+This means we express everything we talk about in this context in the form of models.
+
+But what is a model?
+STOP -- don't run away!
+This question is infamous in the modeling community -- but we have to answer it given our world view as stated above.
+We deliberately don't use that term in our specifications, as it's so overloaded and ambiguous.
+For our purposes, we can always replace "model" with "a bunch of nodes".
+Then the question transforms to: What is a node?
+
+[[notation]]
+== Notation used in this article and diagrams
+We use integer numbers as node ids, and short names for keys of language elements.
+We label boxes in diagrams that represent M1-level nodes with the node id and its classifier, as in `10:Person`.
+Boxes in diagrams for M2-level language elements are labelled only with the classifier, i.e. `Person`.
+Boxes that represent concepts or concept instances use `C` icon, and `@` icon for annotation (instance) boxes.
+
+[[nodes]]
+== Nodes
+A node is the fundamental building block in LionWeb.
+It consists of a unique identifier (**id**), a **classifier** to define the nature / type / class of the node, the node's **parent** node and lists of **properties**, **containments**, **references**, and **annotation instances**, respectively.
+We group a node's properties, containments, and references as the **features** of that node.
+
+Properties are simple values like `name: String`, `age: Integer` or `alive: Boolean`.
+Containments are other nodes that this node is composed of; examples would be `mainEntrance: Door[1]` or `corners: Coordinate[0..\*]`.
+References are other nodes that this node relates to; examples would be `surface: Material[1]` or `stakeholders: Employee[1..*]`.
+Annotation instances are other nodes that represent orthogonal concerns of this node; examples include `ParsedFromLine` or `ReviewComment`.
+
+Let's look into each of these in more detail.
+
+[[node-id]]
+=== Node id
+The id of a node is a string of arbitrary length, consisting of characters `A`-`Z`, `a`-`z`, `0`-`9`, dash `-` and underscore `_`.
+It cannot be empty, and it must be unique for all nodes in the model.
+
+The valid character set coincides to the output of _base64url_-encoding.
+This comes very handy if we had to derive the node id from some other source, like a binary hash, or fully qualified name.
+
+We chose this character set, as it should be compatible with any run-time environment, and avoid any encoding issues.
+We don't limit the id's length, so any kind of other source can be represented -- fully qualified names tend to get very long, for example.
+
+We assume nodes with equal id string to be identical.
+Something went wrong if we had e.g. two different Java objects of class `Node`, but with equal `nodeId` field.
```

Comment:
The word "java" or even programming languages here come out of nowhere. Maybe the stage could be set even earlier with "What is Lionweb" and quickly describing traditional code and model driven DSL code in the first or the "Intended audience" section